LiteCore/Support/Error.cc

```diff
@@ -56,6 +56,87 @@ namespace litecore {
 
     using namespace std;
 
+#ifdef _WIN32
+    static string win32_message(int err) {
+        char buf[1024];
+        buf[0] = '\x0';
+        FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+			nullptr, err, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			buf, sizeof(buf), nullptr);
+        return string(buf);
+    }
+
+    static string cbl_strerror(int err) {
+        if(err < sys_nerr) {
+            // As of Windows 10, only errors 0 - 42 have a message in strerror
+            return strerror(err);
+        }
+
+        if(err >= 10000 && err < 12000) {
+            return win32_message(err);
+        }
+
+        // Hope the POSIX definitions don't change...
+        if(err < 100 || err > 140) {
+            return "Unknown Error";
+        }
+
+        static long wsaEquivalent[] = {
+            WSAEADDRINUSE,
```

Comment:
Could we just map the WSA codes to the corresponding POSIX codes when the error is detected?

Reply:
That's exactly what happens, but now we need a reverse map to get the actual message from the OS.  Otherwise you just end up with a bunch of "Unknown Error" strings.  Alternatively I guess we could look for places that we map to POSIX, grab the message first, and then save the message with the C4Error.